x/concentrated-liquidity/export_test.go

```diff
@@ -258,7 +258,7 @@ func (k Keeper) GetAllIncentiveRecordsForUptime(ctx sdk.Context, poolId uint64,
 	return k.getAllIncentiveRecordsForUptime(ctx, poolId, minUptime)
 }
 
-func (k Keeper) CollectIncentives(ctx sdk.Context, owner sdk.AccAddress, positionId uint64) (sdk.Coins, sdk.Coins, error) {
+func (k Keeper) CollectIncentives(ctx sdk.Context, owner sdk.AccAddress, positionId uint64) (sdk.Coins, sdk.Coins, map[int]sdk.Coins, error) {
```

Comment:
Passing around a map is slightly unwieldy but seemed the most expressive and efficient way to differentiate between forfeited incentives on different accumulators without hard coding values. Happy to replace this with a different approach if anyone sees a better path (or wrap it in a struct for better readability)

Reply:
I generally agree with the approach. Some questions:
1. Do we still need a total forfeited amount returned if we have a more granular breakdown in the returned map?
2. Could we make it a fixed-size slice instead of a map? While the difference is negligible, indexing into a slice is more performant and also removes the risk around non-determinism stemming from maps

Reply:
1. It would technically be a slight optimization to remove it & recompute wherever needed, but this approach is also much more backwards-compatible with tests/queries so I opted to take this approach. If I mispriced this & it's worth the refactoring effort, I'd be happy to revisit
2. Ah I didn't realize that would be more efficient! To clarify, you're suggesting making a `Coins` slice of length `len(SupportedUptimes)` and then placing the values in by `uptimeIndex`?

Reply:
1. I'm more worried about having 2 pieces of data that refer to the same thing but can get out of sync when passed around via return. It's always preferable to minimize spreading across variables referring to the same source. (e.g. a map and a sum of values in the map). This is so that if a map gets updated, we don't have to update the sum. Instead, recompute it on demand, especially, given that the number of elements is small and constant.
2. Yeah, that is what I'm suggestions. And, to be clear, the difference between map vs slice is negligible. However, slice still makes more sense to me since we already have a notion of `uptimeIndex` when reasoning about uptimes and their count always stays the same.

Reply:
Agreed with both of these. Thanks for the input!

Reply:
Noting offline discussion here: decided to keep #1 as is given the simplicity and read-only nature of the returned values. Recomputing would require an API break since `CollectIncentives` is used both as an internal function and a message, and a significant portion of the optimizations for this implementation (e.g. keeping forfeited values scaled) would need to be undone to accommodate this.

It might be worth revisiting this in the future if we feel strong enough about it to take the tradeoffs above, but for the purposes of this PR we can keep them separate.